libbeat/processors/fingerprint/fingerprint_test.go

```diff
@@ -0,0 +1,390 @@
+		Timestamp: time.Now(),
+	}
+
+	tests := []struct {
+		method   string
+		expected string
+	}{
+		{
+			"md5",
+			"4c45df4792f3ef850c928ec5f5232538",
+		},
+		{
+			"sha1",
+			"22f76427d626516d3f7a05785165b99617683b22",
+		},
+		{
+			"sha256",
+			"1208288932231e313b369bae587ff574cd3016a408e52e7128d7bee752674003",
+		},
+		{
+			"sha384",
+			"295adfe0bc03908948e4b0b6a54f441767867e426dda590430459c8a147fbba242a38cba282adee78335b9e08877b86c",
+		},
+		{
+			"sha512",
+			"f50ad51b63c92a0ed0c910527119b81806f3110f0afaa1dcb93506a78371ea761e50c0fc09b08c441d832dd2da1b45e5d8361adfb240e1fffc2695122a23e183",
+		},
+	}
+
+	for _, test := range tests {
+		name := fmt.Sprintf("testing %v fingerprinting method", test.method)
+		t.Run(name, func(t *testing.T) {
+			testConfig, err := common.NewConfigFrom(common.MapStr{
+				"fields": []string{"field1", "field2"},
+				"method": test.method,
+			})
+			assert.NoError(t, err)
+
+			p, err := New(testConfig)
+			assert.NoError(t, err)
+
+			newEvent, err := p.Run(testEvent)
+			assert.NoError(t, err)
+
+			v, err := newEvent.GetValue("fingerprint")
+			assert.NoError(t, err)
+			assert.Equal(t, test.expected, v)
+		})
+	}
+}
+
+func TestSourceFields(t *testing.T) {
+	testEvent := &beat.Event{
+		Fields: common.MapStr{
+			"field1": "foo",
+			"field2": "bar",
+			"nested": common.MapStr{
+				"field": "qux",
+			},
+			"unused_field": "baz",
+		},
+		Timestamp: time.Now(),
+	}
+	expectedFingerprint := "3d51237d384215a6e731f2cc67ead6d7d9a5138377897c8f542a915be3c25bcf"
+
+	tests := []struct {
+		name   string
+		fields []string
+	}{
+		{
+			"order is insignificant",
+			[]string{"field1", "nested.field"},
+		},
+		{
+			"order is insignificant",
+			[]string{"nested.field", "field1"},
+		},
+		{
+			"duplicates are ignored",
+			[]string{"nested.field", "field1", "nested.field"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			testConfig, err := common.NewConfigFrom(common.MapStr{
+				"fields": test.fields,
+				"method": "sha256",
+			})
+			assert.NoError(t, err)
+
+			p, err := New(testConfig)
+			assert.NoError(t, err)
+
+			newEvent, err := p.Run(testEvent)
+			assert.NoError(t, err)
+
+			v, err := newEvent.GetValue("fingerprint")
+			assert.NoError(t, err)
+			assert.Equal(t, expectedFingerprint, v)
+		})
+	}
+}
+
+func TestEncoding(t *testing.T) {
+	testEvent := &beat.Event{
+		Fields: common.MapStr{
+			"field1": "foo",
+			"field2": "bar",
+			"nested": common.MapStr{
+				"field": "qux",
+			},
+			"unused_field": "baz",
+		},
+		Timestamp: time.Now(),
+	}
+
+	tests := []struct {
+		encoding            string
+		expectedFingerprint string
+	}{
+		{
+			"hex",
+			"8934ca639027aab1ee9f3944d4d6bd1e",
+		},
+		{
+			"base32",
+			"RE2MUY4QE6VLD3U7HFCNJVV5DY======",
+		},
+		{
+			"base64",
+			"iTTKY5AnqrHunzlE1Na9Hg==",
+		},
+	}
+
+	for _, test := range tests {
+		name := fmt.Sprintf("testing %v encoding", test.encoding)
+		t.Run(name, func(t *testing.T) {
```

Comment:
Test names can be filtered using regexes. For this I like to give it some structure instead of having full names (assuming that the parent test its name is expressive enough). e.g. just pass encoding as name to `t.Run`. (In case I have multiple parameters or want a name I use `fmt.Sprintf("field=%v" , value)`).

In this case the test name could just be `TestEncoding/base64`. I can run the test using `go test -run Encoding/base64` (the `/` acts as a delimiter.